### PR TITLE
Implement Author, Game class and method

### LINKED ISSUE #16: [4pt] Create a Game class in a separate .rb file.
All Game class properties visible in the diagram should be defined and set up in the constructor method.

Implement methods:
- can_be_archived?() in the Game class
  - should override the method from the parent class
  - should return true if the parent's method returns true AND if 
last_played_at is older than 2 years
  - otherwise, it should return false

### LINKED ISSUE #17: [4pt] Create an Author class with an association to the Item class (in a separate .rb file).
All Author class properties visible in the diagram should be defined and set up in the constructor method.

Implement methods:
- add_item method in the Author class
  - should take an instance of the Item class as an input
  - should add the input item to the collection of items
  - should add self as a property of the item object (by using the correct setter from the item object)

### LINKED ISSUE #18: [2pt] Add unit tests for all implemented methods (Game & Author).

### LINKED ISSUE #19: [2pt] Add Game and Author options.
The following options should be available:
- List of games
- List all authors (e.g. 'Stephen King')
- Add a game

### LINKED ISSUE #20: [2pt] All data should be preserved by saving collections in .json files.
- Game
- Author

### LINKED ISSUE #21: [2pt] Create a schema.sql file with tables that will be analogical to the structure of the classes that you created.
- games table (add all properties and associations from the parent Item class as table columns)
- authors table

### PR DESCRIPTION
**Pull request by desorgui**
## Pull request Changes

- Create a Game class in a separate .rb file.
- Create an Author class with an association to the Item class (in a separate .rb file).
- All Game class properties visible in the diagram were defined in the constructor method.
- All Author class properties visible in the diagram were defined and set up in the constructor method.
- Implement methods:
    - add_item method in the Author class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Game class
        - should override the method from the parent class
        - should return true if parent's method returns true AND if last_played_at is older than 2 years
        - otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options are available:
    - List of games
    - List all authors (e.g. 'Stephen King')
    - Add a game
 - All data are preserved by saving collections in .json files.
 - Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
    - games table (add all properties and associations from the parent Item class as table columns)
    - authors table

### CODE REVIEW

File: methods/author_actions.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#       puts "There isn't any author in our catalog."
+#       sleep(1)
+#     else
+#       @authors.each_with_index do |element, idx|
+#         line = "#{idx + 1}) Author: #{element.first_name} #{element.last_name}"
+#         puts line
+#       end
+#       sleep(2)
+#     end
+#   end
```
Comment (ulisesedu7) [inline methods/author_actions.rb:12]: Here there's commented code.

File: methods/game_actions.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        print line
+      end
+      sleep(2)
+    end
+  end
+
+  def game_create
+    publish_date, last_played_at, multiplayer = game_create_options
+    # title, author_first_name, author_last_name, genre,
+    #  _cover_state, label_color = game_create_options
```
Comment (ulisesedu7) [inline methods/game_actions.rb:17]: Also in this game_actions we have commented code.

**Review by ulisesedu7 — CHANGES_REQUESTED** [review]
# Changes Required 🎯

Hi @desorgui, your code is amazing! Everything is super clean and methods are well distributed across the folders, I believe that's just it. 

The only thing that I have to say is for you to remove the commented code you have in the app, as that would make the code look way cleaner. Then, we would be good to go!

**Review by ulisesedu7 — APPROVED** [review]
Approved Pull Request! 

Good job Desor! We're ready to merge now.

**Comment by David-Verde** [comment]
Status: ✔️ Project Approved ✔️
Now you can merge this branch 🚀

**Pull request merged**